### PR TITLE
Missing BQ SQL definition statements [sc-9930]

### PR DESCRIPTION
**Pull request by alyiwang**
### 🤔 Why?

The BigQuery REST API only returns DDLs for (materialized) views: https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#Table

We need to query against `INFORMATION_SCHEMA.TABLES` to get DDL https://cloud.google.com/blog/topics/developers-practitioners/spring-forward-bigquery-user-friendly-sql

### 🤓 What?

- Fetch table DDL from `INFORMATION_SCHEMA.TABLES`

### 🧪 Tested?

Tested locally against bigquery instance, verified DDL in MCE

### CODE REVIEW

**Review by elic-eon — DISMISSED** [review]

File: metaphor/bigquery/extractor.py [hunk 1]
```diff
@@ -55,17 +55,32 @@ async def extract(self, config: BigQueryRunConfig) -> Collection[ENTITY_TYPES]:
...
+                        ),
+                    )
+                }
+
+                table_ddl = client.query(
+                    f"select table_name, ddl from `{dataset_ref.project}.{dataset_ref.dataset_id}.INFORMATION_SCHEMA.TABLES`"
+                ).result()
+
+                for table_name, ddl in table_ddl:
+                    dataset = datasets[table_name]
```
Comment (mars-lan) [inline metaphor/bigquery/extractor.py:79]: May need to normalize table_name?

File: metaphor/bigquery/extractor.py [hunk 2]
```diff
@@ -55,17 +55,32 @@ async def extract(self, config: BigQueryRunConfig) -> Collection[ENTITY_TYPES]:
...
+                    assert dataset, f"table {table_name} not found for DDL"
```
Comment (mars-lan) [inline metaphor/bigquery/extractor.py:80]: Maybe log instead of throwing exceptions?

File: metaphor/bigquery/extractor.py (hunk 1 above)
Comment (alyiwang) [reply metaphor/bigquery/extractor.py]: good point, updated

File: metaphor/bigquery/extractor.py (hunk 2 above)
Comment (alyiwang) [reply metaphor/bigquery/extractor.py]: updated

**Review by mars-lan — APPROVED** [review]

**Pull request merged**